src/core/service.c

```diff
@@ -2648,7 +2648,7 @@ static void service_run_next_control(Service *s) {
         r = service_spawn(s,
                           s->control_command,
                           timeout,
-                          EXEC_APPLY_SANDBOXING|EXEC_APPLY_CHROOT|EXEC_IS_CONTROL|
+                          EXEC_APPLY_SANDBOXING|EXEC_APPLY_CHROOT|EXEC_IS_CONTROL|EXEC_WRITE_CREDENTIALS|
```

Comment:
this would also load the creds on exec stop. please don't. it's ok to load them on "positive" states, but let's avoid negative states.

i.e. you can reause the if check above that figures out the timeouts: it's the same cases where we should write creds.

Reply:
fixed